test/luatest_helpers/vtest.lua

```diff
@@ -601,6 +602,128 @@ local function drop_instance(g, instance)
     g[instance.alias] = nil
 end
 
+--
+-- Wait until the member of the tab (table) becomes not equal to nil.
+-- As we want to have a 'pointer' to the member and not just the copy
+-- of nil, we pass table and the expected member's name.
+--
+local function wait_for_not_nil(tab, member, opts)
+    opts = opts or {}
+    t.assert_equals(type(tab), 'table')
+    t.helpers.retrying({timeout = opts.timeout or wait_timeout,
+                        delay = busy_step}, function()
+        if tab[member] == nil then
+            if opts.on_yield then
+                opts.on_yield()
+            end
+            error(string.format('timed out: %s is still nil', member))
+        end
+    end)
+end
+
+--------------------------------------------------------------------------------
+-- Service info helpers
+--------------------------------------------------------------------------------
+
+--
+-- Wait for the status of the service to be equal to the one user expects.
+-- The function assumes, that the current status is not the one user want
+-- to see (requested status must have another status_idx).
+--
+-- Following opts are accepted: opts.timeout and opts.on_yield
+--
+local function service_wait_for_new_status(service, status, opts)
+    opts = opts or {}
+    local first_status_idx = service.status_idx
+    t.helpers.retrying({timeout = opts.timeout or wait_timeout,
+                        delay = busy_step}, function()
+        if first_status_idx ~= service.status_idx and
+           service.status == status then
+            return
+        end
+        if opts.on_yield then
+            opts.on_yield()
+        end
+        error(string.format('waiting for status "%s" timed out: ' ..
+                            'last status is "%s" with status_idx %d',
+                            status, service.status, service.status_idx))
+    end)
+end
+
+--
+-- Wait until the error, passed to the function's argument, occurs.
+-- The error must have new status_idx different from the current one.
+--
+local function service_wait_for_new_error(service, error, opts)
+    log.info('Waiting for new error "%s" of "%s" service',
+             error, service.name)
+    while not service.status == 'error' or not
+          string.match(service.error, error) do
```

Comment:
The function is supposed to skip the first error by status idx, but in this implementation it will be accepted - you will not enter `while` if status is already `error` and the error message matches.

Reply:
You're right fixed